modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,147 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/nw-configuring-ingress-controller-endpoint-publishing-strategy.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-controller-nodeportservice-projects_{context}"]
+= Adding a single NodePort service to an Ingress Controller
+
+Instead of creating a `NodePort`-type `Service` for each project, you can create a custom Ingress Controller to use the `NodePortService` endpoint publishing strategy. To prevent port conflicts, consider this configuration for your Ingress Controller when you want to apply a set of routes, through Ingress sharding, to nodes that might already have a `HostNetwork` Ingress Controller.  
+
+Before you set a `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Ingress Controller. For more information about the required DNS records for routes, see "User-provisioned DNS requirements".
+* You must expose a route for your service and specify the `--hostname` argument for your custom Ingress Controller domain.
+* Ensure that all Ingress Controller routes can access your {product-title} cluster through the host IP address assigned to each node.
+* You must use the `curl` command to transfer data to the `NodePort`-type `Service` in the custom Ingress Controller.
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the `IngressController` object
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: <custom_name> <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: <domain> <2>
+    nodePlacement:
+      nodeSelector:
+        matchLabels:
+          ingresscontroller: <custom_name> <3>
```

Comment:
note: I used "ingresscontroller" here as the label key but you get to chose what the whole label pair is.  So I could have called it this:
ic:  <custom_name>
just like with the namespaceSelector.  So essentially:
<key>: <value.

Reply:
:heavy_check_mark: